File: app/src/main/java/ro/scoalainformala/trips/Trip/TripRepository.java
package ro.scoalainformala.trips.Trip;

import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

public class TripRepository {

    private TripDao tripDao;

    public TripRepository(TripDao tripDao){
        this.tripDao = tripDao;
    }

    LiveData<List<Trip>> getTrips(){
        return tripDao.getAlphabetizedTrips();
    }

    public void insert(Trip trip){
        TripDatabase.EXECUTOR.execute(() -> tripDao.insert(trip));
    }

    public void update(Trip trip){
        TripDatabase.EXECUTOR.execute(() -> tripDao.update(trip));
    }

    public LiveData<List<Trip>> getAllFavouriteTrips(){return tripDao.getAllFavouriteTrips();}

//    public void update(Trip trip) {
//        new UpdateTripAsyncTask(tripDao).execute(trip);
//    }
//
//    private static class UpdateTripAsyncTask extends AsyncTask<Trip, Void, Void> {
//        private TripDao tripDao;
//
//        private UpdateTripAsyncTask(TripDao tripDao) {
//            this.tripDao = tripDao;
//        }
//
//        @Override
//        protected Void doInBackground(Trip... trips) {
//            tripDao.update(trips[0]);
//            return null;
//        }
//    }
}

File: app/src/main/java/ro/scoalainformala/trips/Trip/TripRepository.kt
package ro.scoalainformala.trips.Trip

import androidx.lifecycle.LiveData

class TripRepository(private val tripDao: TripDao) {
    val trips: LiveData<List<Trip>>
        get() = tripDao.alphabetizedTrips

    fun insert(trip: Trip) {
        TripDatabase.Companion.EXECUTOR.execute(Runnable { tripDao.insert(trip) })
    }

    fun update(trip: Trip) {
        TripDatabase.Companion.EXECUTOR.execute(Runnable { tripDao.update(trip) })
    }

    val allFavouriteTrips: LiveData<List<Trip>>
        get() = tripDao.allFavouriteTrips
}
